editor/components/post-featured-image/check.js

```diff
@@ -1,15 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
 import PostTypeSupportCheck from '../post-type-support-check';
 import ThemeSupportCheck from '../theme-support-check';
 
 function PostFeaturedImageCheck( props ) {
+	const { postType } = props;
 	return (
-		<ThemeSupportCheck supportKeys="post-thumbnails">
+		<ThemeSupportCheck supportKeys="post-thumbnails" postType={ postType } >
```

Comment:
Should it be the responsibility of the consumer to pass in the `postType` prop? Or should `ThemeSupportCheck` manage this itself?

Otherwise, how would we document what this prop is, and when it's to be passed? It's not really a "thing" so far as theme support goes. It's just one of many possible [miscellaneous `...$args`](https://developer.wordpress.org/reference/functions/add_theme_support/#parameters) which could be passed.

My first inclination was: Maybe we don't have `ThemeSupportCheck` and we just make `PostFeaturedImageCheck` a simple series of higher-order components using `withSelect` (and `getEditedPostAttribute`, `getThemeSupports`, `getPostTypeSupports`) and `ifCondition`. But there is a point to be made about consolidating where we consider something like "post thumbnails is possibly an array".

Which then makes me wonder: Should this be part of a selector? Something like `hasThemeSupport` which includes as part of its internal logic to consider the key and any special considerations to make.

Reply:
Updated [per our Slack conversation](https://wordpress.slack.com/archives/C02QB2JS7/p1530126701000439) 80a27f18a